pkg/status/status.go

```diff
@@ -200,3 +227,71 @@ func (s *status) updateStatus() {
 		log.Fatalf("Error %v updating ClusterOperator", err)
 	}
 }
+
+// syncChannelConsumer will listen on the sync channel and update the status'
+// syncs and successfulSyncs fields until the stopCh is closed.
+func syncChannelConsumer(instance *status, stopCh <-chan struct{}) {
```

Comment:
you can use `select` to express what `syncChannelConsumer` does in a concise manner.
```
func syncChannelConsumer(instance *status, stopCh <-chan struct{}) {
  for {
    select {
    case <-stopCh:
        return
     case err := <-syncCh
       ...
    }
  }
}
```